app/models/concerns/audit_extensions.rb

```diff
@@ -222,7 +222,7 @@ def set_taxonomy_for(taxonomy)
         auditable.send(taxonomy_attribute), audited_changes[taxonomy_attribute.to_s]
       ].flatten.compact.uniq)
     elsif auditable.respond_to?(taxonomy_attribute_plural)
-      send("#{taxonomy_attribute_plural}=", auditable.send(taxonomy_attribute_plural))
+      send("#{taxonomy_attribute_plural}=", auditable.send(taxonomy_attribute_plural).compact.uniq)
```

Comment:
Two questions:
1) Is `compact`required here? I mean I am not sure if the values are `nil` in the database.  If not then we can remove it from here. Is it possible to have a unique constraint on the database field itself?
2) Can the values in the database be a empty string? cause if it can be then we are not attending to that condition here.

Reply:
> Is compact required here? I mean I am not sure if the values are nil in the database. If not then we can remove it from here. Is it possible to have a unique constraint on the database field itself?

No need of compact. Removed!

> Can the values in the database be a empty string? cause if it can be then we are not attending to that condition here.

Not sure, I am getting this. But do you any specific scenario where {location/orgnization}_ids include empty string?

Reply:
AFAIK, taxonomies get default values so they must not have a nil or empty value associated to them. Ignore the last part.